utils/apps.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppBinding} from 'mattermost-redux/types/apps';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+export function appsEnabled(state: GlobalState) {// eslint-disable-line @typescript-eslint/no-unused-vars
+    // TODO uncomment when featur flag is in place
+    //return getConfig(state)?.['FeatureFlagApps' as keyof Partial<ClientConfig>];
+    return true;
+}
+
+export function fillBindingsInformation(binding: AppBinding) {
+    binding.bindings?.forEach((b) => {
+        // Propagate id down if not defined
+        if (!b.app_id) {
+            b.app_id = binding.app_id;
+        }
+
+        // Compose location
+        b.location = binding.location + '/' + b.location;
+
+        // Propagate call down if not defined
+        if (!b.call) {
+            b.call = binding.call;
+        }
+
+        fillBindingsInformation(b);
```

Comment:
In what circumstances do we have more than two levels of bindings?

Reply:
With commands we have an undefined amount of level of bindings.
In message attachments, at the moment we have 3 levels: Attachment -> Menu Binding -> Option.
Potentially (depending on UX) we may have an undefined amount of levels too: Attachment -> Other -> Menu item -> Submenu item -> Subsubmenu item...
Similar situation with channel header and post menu items:
ChannelHeader -> Element -> SubMenu -> Subsubmenu...